pkg/partition/partition.go

```diff
@@ -0,0 +1,111 @@
+/*
+Copyright 2020 The OpenEBS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package partition
+
+import (
+	"fmt"
+
+	"github.com/diskfs/go-diskfs"
+	"github.com/diskfs/go-diskfs/partition/gpt"
+	"k8s.io/klog"
+)
+
+// TODO this package needs to be upstreamed to diskfs/go-diskfs
+
+const (
+	// BytesRequiredForGPTHeader is the total bytes required to store the GPT header
+	// 128 bytes are required per partition, total no of partition supported by GPT = 128
+	// Therefore, total bytes = 128*128
+	BytesRequiredForGPTHeader = 16384
+
+	// GPTPartitionStartByte is the byte on the disk at which the first partition starts.
+	// Normally partition starts at 1MiB, (as done by fdisk utility). This is done to
+	// align the partition start to physical block sizes on the disk.
+	GPTPartitionStartByte = 1048576
+)
+
+// Disk struct represents a disk which needs to be partitioned
+type Disk struct {
+	// DevPath is the /dev/sdX entry of the disk
+	DevPath string
+	// DiskSize is size of disk in bytes
+	DiskSize uint64
+	// LogicalBlockSize is the block size of the disk normally 512 or 4k
+	LogicalBlockSize uint64
+
+	table *gpt.Table
+}
+
+// ApplyPartitionTable applies the partition table to the given disk.
```

Comment:
Can we make these methods private to this function and just expose a single method that partitions the whole disk using the internal methods? Also, make this method idempotent so that if a partition exists - it doesn't recreate it. 

New methods can be added later that helps in creating additional partitions.

Reply:
Why do we have to do in such a way @kmova ? 
That will be abstracting too much details right? we can make the function to error out if a partition already exists.

Reply:
We can create a method over it , instead of calling 3 separate methods. But still I am not in the strong opinion of making a single methods public and rest of them private.

Reply:
How do we ensure that partitioning gets called only once and have the validation that partition should be created only if the device doesn't already have one?

We can leave it up to the caller, but what if there are multiple callers - possibly due to a bug in the future contributions. 

It helps to lock down as much as possible so that callers don't make mistakes.

Reply:
Done.